code/game/objects/effects/effect_system/effects_chem_smoke.dm

```diff
@@ -0,0 +1,123 @@
+	var/matrix/first = matrix()
+	var/matrix/second = matrix()
+	var/matrix/third = matrix()
+	var/first_scale = 0.1
+	var/second_scale = 5
+	var/third_scale = 2
+	var/spread_amount = 96
+
+/obj/effect/particle_effect/chem_smoke/New(location, chem_color)
+	..()
+	color = chem_color
+	pixel_x += -16 + rand(-3, 3)
+	pixel_y += -16 + rand(-3, 3)
+
+	first = turn(first, rand(-90, 90))
+	first.Scale(first_scale, first_scale)
+	transform = first
+
+	second = first
+	second.Scale(second_scale, second_scale)
+
+	third.Scale(third_scale, third_scale)
+
+	animate(src,transform = second, time = 5, alpha = 200)
+	animate(transform = third, time = 20, pixel_y = rand(-spread_amount, spread_amount), pixel_x = rand(-spread_amount, spread_amount), alpha = 1)
+	QDEL_IN(src, 26)
+
+
+/obj/effect/particle_effect/chem_smoke/small
+	first_scale = 0.05
+	second_scale = 2.5
+	third_scale = 1
+	spread_amount = 48
+
+
+/datum/effect_system/smoke_spread/chem
+	var/obj/chemholder
+	var/list/smoked_atoms = list()
+
+/datum/effect_system/smoke_spread/chem/New()
+	..()
+	chemholder = new/obj()
+	var/datum/reagents/R = new/datum/reagents(500)
+	chemholder.reagents = R
+	R.my_atom = chemholder
+
+/datum/effect_system/smoke_spread/chem/Destroy()
+	QDEL_NULL(chemholder)
+	smoked_atoms.Cut()
+	return ..()
+
+/datum/effect_system/smoke_spread/chem/set_up(datum/reagents/carry = null, loca, silent = FALSE)
+	if(isturf(loca))
+		location = loca
+	else
+		location = get_turf(loca)
+	carry.copy_to(chemholder, carry.total_volume)
+	if(!silent)
+		var/contained = ""
+		for(var/reagent in carry.reagent_list)
+			contained += " [reagent] "
+		if(contained)
+			contained = "\[[contained]\]"
+		var/area/A = get_area(location)
+
+		var/where = "[A.name] | [location.x], [location.y]"
+		var/whereLink = "<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[location.x];Y=[location.y];Z=[location.z]'>[where]</a>"
+
+		if(carry && carry.my_atom)
+			if(carry.my_atom.fingerprintslast)
+				var/mob/M = get_mob_by_key(carry.my_atom.fingerprintslast)
+				var/more = ""
+				if(M)
+					more = " "
+				msg_admin_attack("A chemical smoke reaction has taken place in ([whereLink])[contained]. Last associated key is [carry.my_atom.fingerprintslast][more].", ATKLOG_FEW)
+				log_game("A chemical smoke reaction has taken place in ([where])[contained]. Last associated key is [carry.my_atom.fingerprintslast].")
+			else
+				msg_admin_attack("A chemical smoke reaction has taken place in ([whereLink]). No associated key.", ATKLOG_FEW)
+				log_game("A chemical smoke reaction has taken place in ([where])[contained]. No associated key.")
+		else
+			msg_admin_attack("A chemical smoke reaction has taken place in ([whereLink]). No associated key. CODERS: carry.my_atom may be null.", ATKLOG_FEW)
+			log_game("A chemical smoke reaction has taken place in ([where])[contained]. No associated key. CODERS: carry.my_atom may be null.")
+
+
+/datum/effect_system/smoke_spread/chem/start(effect_range = 2)
+	set waitfor = FALSE
```

Comment:
this effect lasts about ~10 seconds, we need this to prevent some things from halting.

For instance, blobspores burst when they die and release smoke and get deleted. We don't want the blob spore to hang around for the duration of the smoke before it gets deleted.